cmd/kubeadm/app/util/staticpod/utils_test.go

```diff
@@ -209,125 +209,125 @@ func TestEtcdProbe(t *testing.T) {
 			cfg: &kubeadmapi.MasterConfiguration{
 				Etcd: kubeadmapi.Etcd{
 					Local: &kubeadmapi.LocalEtcd{
 						ExtraArgs: map[string]string{
-							"listen-client-urls": "http://1.2.3.4:2379,http://4.3.2.1:2379"},
+							"listen-client-urls": "http://[::0:0]:5"},
 					},
 				},
 			},
 			component: kubeadmconstants.Etcd,
-			port:      1,
-			certsDir:  "secretsA",
-			cacert:    "ca1",
-			cert:      "cert1",
-			key:       "key1",
-			expected:  "ETCDCTL_API=3 etcdctl --endpoints=https://[1.2.3.4]:1 --cacert=secretsA/ca1 --cert=secretsA/cert1 --key=secretsA/key1 get foo",
+			port:      5,
+			certsDir:  "secrets_E",
+			cacert:    "ca5",
+			cert:      "cert5",
+			key:       "key5",
+			expected:  "ETCDCTL_API=3 etcdctl --endpoints=https://[::1]:5 --cacert=secrets_E/ca5 --cert=secrets_E/cert5 --key=secrets_E/key5 get foo",
 		},
 		{
-			name: "valid etcd probe using listen-client-urls IPv6 addresses",
+			name: "valid etcd probe using listen-client-urls unspecified IPv6 address (no zeros)",
 			cfg: &kubeadmapi.MasterConfiguration{
 				Etcd: kubeadmapi.Etcd{
 					Local: &kubeadmapi.LocalEtcd{
 						ExtraArgs: map[string]string{
-							"listen-client-urls": "http://[2001:db8::1]:2379,http://[2001:db8::2]:2379"},
+							"listen-client-urls": "http://[::]:6"},
 					},
 				},
 			},
 			component: kubeadmconstants.Etcd,
-			port:      1,
-			certsDir:  "secretsB",
-			cacert:    "ca2",
-			cert:      "cert2",
-			key:       "key2",
-			expected:  "ETCDCTL_API=3 etcdctl --endpoints=https://[2001:db8::1]:1 --cacert=secretsB/ca2 --cert=secretsB/cert2 --key=secretsB/key2 get foo",
+			port:      6,
+			certsDir:  "secrets_F",
+			cacert:    "ca6",
+			cert:      "cert6",
+			key:       "key6",
+			expected:  "ETCDCTL_API=3 etcdctl --endpoints=https://[::1]:6 --cacert=secrets_F/ca6 --cert=secrets_F/cert6 --key=secrets_F/key6 get foo",
 		},
 		{
-			name: "valid IPv4 etcd probe using hostname for listen-client-urls",
+			name: "valid etcd probe using listen-client-urls unspecified IPv4 address",
 			cfg: &kubeadmapi.MasterConfiguration{
 				Etcd: kubeadmapi.Etcd{
 					Local: &kubeadmapi.LocalEtcd{
 						ExtraArgs: map[string]string{
-							"listen-client-urls": "http://localhost:2379"},
+							"listen-client-urls": "http://0.0.0.0:7"},
 					},
 				},
 			},
 			component: kubeadmconstants.Etcd,
-			port:      1,
-			certsDir:  "secretsC",
-			cacert:    "ca3",
-			cert:      "cert3",
-			key:       "key3",
-			expected:  "ETCDCTL_API=3 etcdctl --endpoints=https://[127.0.0.1]:1 --cacert=secretsC/ca3 --cert=secretsC/cert3 --key=secretsC/key3 get foo",
+			port:      7,
+			certsDir:  "secrets_G",
+			cacert:    "ca7",
+			cert:      "cert7",
+			key:       "key7",
+			expected:  "ETCDCTL_API=3 etcdctl --endpoints=https://[127.0.0.1]:7 --cacert=secrets_G/ca7 --cert=secrets_G/cert7 --key=secrets_G/key7 get foo",
 		},
 	}
 	for _, rt := range tests {
```

Comment:
would you be cool adding a name to each of these test cases and running them with `t.Run()` inside the `for` loop?

using `t.Run` lets us pinpoint exactly which test in the table failed and adds some documentation about what each case is testing (because we must provide a name).

Reply:
I'll second this, troubleshooting failed tests without names is painful.